packages/sdk/src/actions/bridge.ts

```diff
@@ -1,71 +1,167 @@
+ * @param data.amount The amount to bridge
+ * @param data.currency The currency to bridge
+ * @param data.recipient The address that will receive the bridge
  * @param data.wallet Wallet object that adheres to the AdaptedWakket interface or a viem WalletClient
- * @param data.originChainId The chain to pay the solver on
  * @param data.precheck Set to true to skip executing steps and just to get the initial steps required
+ * @param data.options - {@link BridgeBodyOptions}
  * @param data.onProgress Callback to update UI state as execution progresses
  */
+
 export async function bridge(data: BridgeActionParameters) {
-  const { to, wallet, value, precheck, ...proxiedData } = data
+  const {
+    chainId,
+    toChainId,
+    wallet,
+    amount,
+    currency = 'eth',
+    recipient,
+    onProgress = () => {},
+    precheck,
+    depositGasLimit,
+    options,
+  } = data
+  const client = getClient()
+
+  if (!client.baseApiUrl || !client.baseApiUrl.length) {
+    throw new ReferenceError('RelayClient missing api url configuration')
+  }
 
   let adaptedWallet: AdaptedWallet | undefined
   let caller: string | undefined
-
   if (wallet) {
     adaptedWallet = isViemWalletClient(wallet)
       ? adaptViemWallet(wallet as WalletClient)
       : wallet
     caller = await adaptedWallet.address()
   }
 
+  // Ensure wallet is provided when precheck is false or undefined
   if (!precheck && !adaptedWallet) {
     throw new Error(
-      'Wallet is required when precheck is false or not provided.',
+      'Wallet is required when precheck is false or not provided.'
     )
   }
 
-  return call({
-    ...proxiedData,
-    wallet: adaptedWallet as AdaptedWallet,
-    txs: [
-      {
-        to: to ?? caller ?? zeroAddress,
-        value,
-      },
-    ],
-    precheck,
-  })
+  try {
+    const data: BridgeBody = {
+      user: caller || zeroAddress,
+      recipient: recipient,
+      originChainId: chainId,
+      destinationChainId: toChainId,
+      currency,
+      amount,
+      source: client.source || undefined,
+      ...options,
+    }
+
+    const request: AxiosRequestConfig = {
+      url: `${client.baseApiUrl}/execute/bridge`,
+      method: 'post',
+      data,
+    }
+
+    if (precheck) {
+      const res = await axios.request(request)
+      if (res.status !== 200)
+        throw new APIError(res?.data?.message, res.status, res.data)
+      const data = res.data as Execute
+      onProgress(data['steps'], data['fees'], data['breakdown'])
+      return data
+    } else {
+      if (!adaptedWallet) {
+        throw new Error('AdaptedWallet is required to execute steps')
+      }
+
+      await executeSteps(
+        chainId,
+        request,
+        adaptedWallet,
+        (
+          steps: Execute['steps'],
+          fees: Execute['fees'],
+          breakdown: Execute['breakdown']
+        ) => {
+          let currentStep: NonNullable<Execute['steps']>['0'] | null = null
```

Comment:
I wonder if there's any way we can clean this up with some utility functions, this code is also repeated in the call action right?